tests/ProjectIdProviderTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+namespace Google\Auth\Tests;
+
+use Google\Auth\CredentialsLoader;
+use Google\Auth\ProjectIdProvider;
+use PHPUnit\Framework\TestCase;
+
+class ProjectIdProviderShellMock extends ProjectIdProvider
```

Comment:
If you'd like to stick with testing private methods, WDYT about a little helper on the shell class you have here which will encapsulate the reflection/setAccessibility calls?